src/Compilers/CSharp/Portable/Symbols/Source/ParameterHelpers.cs

```diff
@@ -600,5 +604,33 @@ private static RefKind GetModifiers(SyntaxTokenList modifiers, out SyntaxToken r
 
             return refKind;
         }
+
+        internal static void AddNullCheckingErrorsToParameter(DiagnosticBag diagnostics, ParameterSymbol parameter, bool isLambdaOrLocalFunction)
+        {
+            if (parameter.IsNullChecked)
+            {
+                Location location = parameter.Locations.FirstOrNone();
+                if (Binder.GetWellKnownTypeMember(parameter.DeclaringCompilation, WellKnownMember.System_ArgumentNullException__ctorString, out DiagnosticInfo diag) is null)
+                {
+                    diagnostics.Add(diag, location);
+
+                }
+                if (parameter.Type.IsValueType)
+                {
+                    if (!parameter.Type.IsNullableTypeOrTypeParameter())
+                    {
+                        diagnostics.Add(ErrorCode.ERR_NonNullableValueTypeIsNullChecked, location, parameter);
+                    }
+                    else
+                    {
+                        diagnostics.Add(ErrorCode.WRN_NullCheckingOnNullableValueType, location, parameter);
+                    }
+                }
+                if (isLambdaOrLocalFunction && parameter.ExplicitDefaultConstantValue?.IsNull == true)
+                {
+                    diagnostics.Add(ErrorCode.WRN_NullCheckedHasDefaultNull, location, parameter.Name);
```

Comment:
>diagnostics.Add(ErrorCode.WRN_NullCheckedHasDefaultNull, location, parameter.Name); [](start = 20, length = 83)

Is this case ever hit? Lambda parameters cannot have default values it is only `LambdaSymbol` that passes `isLambdaOrLocalFunction: true`.

Please consider removing block and `isLambdaOrLocalFunction` parameter if not used.